provider/core/src/test/java/org/arquillian/pact/provider/core/InteractionRunnerTest.java

```diff
@@ -87,6 +88,25 @@ public void should_throw_exception_when_no_target() {
 
     }
 
+    @org.junit.Test
+    public void should_throw_exception_when_state_param_is_not_empy_nor_map() {
+        when(test.getTestClass()).thenReturn(new TestClass(PactProviderWithWrongStateMethod.class));
+        PactProviderWithWrongStateMethod pactDefinition = new PactProviderWithWrongStateMethod();
+        when(test.getTestInstance()).thenReturn(pactDefinition);
+
+        InteractionRunner interactionRunner = new InteractionRunner();
+        interactionRunner.pactsInstance = pactsInstance;
+        interactionRunner.targetInstance = () -> target;
+
+        try {
+            interactionRunner.executePacts(eventContext);
+            fail("Exception should be thrown");
+        } catch(IllegalArgumentException e) {
+            assertThat(e).hasMessage("Method stateMethod should take only a single Map parameter");
```

Comment:
Well I don't know if this is more simple hehehe but at least it uses the assertj correctly which is great